### PR TITLE
eManager cleanup

### PR DESCRIPTION
**Pull request by koseoyoung**
fixed bugs in the endorsementManager cleanup, and added unit tests

### CODE REVIEW

File: consensus/scheme/rolldpos/endorsementmanager.go [hunk 1]
```diff
@@ -368,16 +368,13 @@ func (m *endorsementManager) AddVoteEndorsement(
 }
 
 func (m *endorsementManager) Cleanup(timestamp time.Time) error {
-	cleaned := &endorsementManager{
-		eManagerDB:  m.eManagerDB,
-		collections: map[string]*blockEndorsementCollection{},
-	}
+	cleanedCollections := map[string]*blockEndorsementCollection{}
```
Comment (dustinxie) [inline consensus/scheme/rolldpos/endorsementmanager.go]: could you just use the change in
https://github.com/iotexproject/iotex-core/commit/f491a23a77c0e47a2ea92344b189b354592835ab

that is what's being tested on nightly-0 now

File: consensus/scheme/rolldpos/endorsementmanager_test.go [hunk 2]
```diff
@@ -125,15 +125,39 @@ func TestEndorsementManager(t *testing.T) {
 	l := em.Log(log.L(), nil)
 	require.NotNil(l)
 	l.Info("test output")
-	err = em.Cleanup(time.Now().Add(time.Second * 10 * -1))
+
+	cv2 := NewConsensusVote(blkHash[:], LOCK)
+	require.NotNil(cv2)
+	end2 := endorsement.NewEndorsement(time.Now(), b.PublicKey(), []byte("456"))
+	require.NoError(em.AddVoteEndorsement(cv2, end2))
+	l = em.Log(log.L(), nil)
```
Comment (CoderZhi) [inline consensus/scheme/rolldpos/endorsementmanager_test.go]: do we need to regenerate l?

File: consensus/scheme/rolldpos/endorsementmanager_test.go [hunk 3]
```diff
@@ -125,15 +125,39 @@ func TestEndorsementManager(t *testing.T) {
 	l := em.Log(log.L(), nil)
 	require.NotNil(l)
 	l.Info("test output")
-	err = em.Cleanup(time.Now().Add(time.Second * 10 * -1))
+
+	cv2 := NewConsensusVote(blkHash[:], LOCK)
+	require.NotNil(cv2)
+	end2 := endorsement.NewEndorsement(time.Now(), b.PublicKey(), []byte("456"))
```
Comment (CoderZhi) [inline consensus/scheme/rolldpos/endorsementmanager_test.go]: we shouldn't use time.Now

File: consensus/scheme/rolldpos/endorsementmanager.go (hunk 1 above)
Comment (koseoyoung) [reply consensus/scheme/rolldpos/endorsementmanager.go]: oh ok !!

**Pull request merged**